src/app/pages/communication/communication.page.ts

```diff
@@ -1,30 +1,354 @@
 import { TestFlowPageNames } from '@pages/page-names.constants';
 
+interface CommunicationPageState {
+  candidateName$: Observable<string>;
+  candidateUntitledName$: Observable<string>;
+  candidateDriverNumber$: Observable<string>;
+  candidateProvidedEmail$: Observable<string>;
+  communicationEmail$: Observable<string>;
+  communicationType$: Observable<string>;
+  candidateAddress$: Observable<Address>;
+  conductedLanguage$: Observable<string>;
+  testCategory$: Observable<CategoryCode>;
+}
+
 @Component({
   selector: 'app-communication',
   templateUrl: './communication.page.html',
   styleUrls: ['./communication.page.scss'],
 })
-export class CommunicationPage implements OnInit {
+export class CommunicationPage extends PracticeableBasePageComponent implements OnInit {
+
+  static readonly providedEmail: string = 'Provided';
+  static readonly updatedEmail: string = 'Updated';
+  static readonly email: CommunicationMethod = 'Email';
+  static readonly post: CommunicationMethod = 'Post';
+  static readonly notProvided: CommunicationMethod = 'Not provided';
+  static readonly welshLanguage: ConductedLanguage = 'Cymraeg';
+  static readonly englishLanguage: ConductedLanguage = 'English';
+
+  form: FormGroup;
+  subscription: Subscription;
+  emailType: string;
+  pageState: CommunicationPageState;
+  candidateProvidedEmail: string;
+  communicationEmail: string;
+  communicationType: CommunicationMethod;
+  merged$: Observable<string | boolean>;
+  testCategory: CategoryCode;
 
   constructor(
-    private navController: NavController,
+    public platform: Platform,
+    public router: Router,
+    public authenticationProvider: AuthenticationProvider,
+    store$: Store<StoreModel>,
     public routeByCat: RouteByCategoryProvider,
-  ) { }
+    public deviceAuthenticationProvider: DeviceAuthenticationProvider,
+    private translate: TranslateService,
+  ) {
+    super(platform, router, authenticationProvider, store$);
+    this.form = new FormGroup(this.getFormValidation());
+  }
+
+  ngOnInit(): void {
+    super.ngOnInit();
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+
+    this.pageState = {
+      candidateName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateName),
+      ),
+      candidateUntitledName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getUntitledCandidateName),
+      ),
+      candidateDriverNumber$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateDriverNumber),
+        map(formatDriverNumber),
+      ),
+      candidateProvidedEmail$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateEmailAddress),
+        take(1),
+      ),
+      communicationEmail$: this.store$.pipe(
```

Comment:
can we pipe from currentTest rather than from store here?

Reply:
was taken directly from des3, but can take a look